dungeon/src/contrib/crafting/Crafting.java

```diff
@@ -80,60 +80,33 @@ public static void clearRecipes() {
    * <p>If the program is compiled to a jar file, recipes will be loaded from within the jar file.
    */
   public static void loadRecipes() {
-    if (Objects.requireNonNull(Crafting.class.getResource("/recipes"))
-        .toString()
-        .startsWith("jar:")) {
-      loadFromJar();
-    } else {
-      loadFromFile();
-    }
-  }
-
-  /** Load recipes if the program was started from a jar file. */
-  private static void loadFromJar() {
+    final String dirName = "recipes/";
     try {
-      String path =
-          new File(Objects.requireNonNull(Game.class.getResource("")).getPath())
-              .getParent()
-              // for windows
-              .replaceAll("(!|file:\\\\)", "")
-              // for unix/macos
-              .replaceAll("(!|file:)", "");
-      JarFile jar = new JarFile(path);
-      Enumeration<JarEntry> entries = jar.entries();
-      while (entries.hasMoreElements()) {
-        JarEntry entry = entries.nextElement();
-        if (entry.getName().startsWith("recipes") && entry.getName().endsWith(".recipe")) {
-          LOGGER.info("Load recipe: " + entry.getName());
-          Recipe r =
-              parseRecipe(Game.class.getResourceAsStream("/" + entry.getName()), entry.getName());
-          if (r != null) Crafting.RECIPES.add(r);
-        }
-      }
-    } catch (IOException e) {
+      // Walk through the 'recipes' directory and load all recipes.
+      Files.walkFileTree(
+          Paths.get(
+              Objects.requireNonNull(
+                      Thread.currentThread().getContextClassLoader().getResource(dirName))
+                  .toURI()),
+          new SimpleFileVisitor<>() {
+            @Override
+            public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs)
+                throws IOException {
+              if (Files.isRegularFile(file) && file.toString().endsWith(".recipe")) {
```

Comment:
Warum `isRegularFile`? Diese Methode wird lt. Doku nur für Files aufgerufen, nicht für Verzeichnisse. Und Sym-Links oder Pipes o.ä. haben wir nicht in unseren Asset-Ordnern?

Reply:
done in [18abe0d](https://github.com/Dungeon-CampusMinden/Dungeon/pull/1366/commits/18abe0d56fa6b873940cd09593f95a82de86f2ba)